src/api/lib/backend/api/build_results/binaries.rb

```diff
@@ -64,16 +64,14 @@ def self.wipe_published_locked(project_name)
         # the name of the binary as keys and the architectures as the value
         def self.transform_binary_packages_response(response)
           list = {}
+          list.default = []
           parsed_response = Xmlhash.parse(response)
           return list if parsed_response.blank?
           packages = [parsed_response["packages"]].flatten
           packages.each do |build|
             architectures = [build["arch"]].flatten
             packages = [build["name"]].flatten
```

Comment:
Please rename the variable to ``package_names``
1) It holds the names and not the packages
2) We already have a packages variable in line 70